File: Arashyn-BE/src/main/java/com/arashi/edu/arashynbe/features/playground/grammar/dto/response/GrammarListResponse.java
package com.arashi.edu.arashynbe.features.playground.grammar.dto.response;

import java.util.List;

public record GrammarListResponse(

        List<GrammarSummaryResponse> items,

        Integer page,

        Integer size,

        Integer totalPages,

        Long totalElements,

        Boolean hasNext,

        Boolean hasPrevious

) {
}

File: Arashyn-FE/src/mocks/types/response/grammar_list_response.ts
import type {GrammarResponse} from "@/entities/grammar/grammar_types.ts";

export type GrammarListResponse = {
    items: GrammarResponse[];
    page: number;
    size: number;
    totalPages: number;
    totalElements: number;
    hasNext: boolean;
    hasPrevious: boolean;
};